src/zenml/zen_stores/rest_zen_store.py

```diff
@@ -406,13 +398,47 @@ def expand_certificates(self) -> None:
             with open(self.verify_ssl, "r") as f:
                 self.verify_ssl = f.read()
 
+    @model_validator(mode="before")
+    @classmethod
+    @before_validator_handler
+    def _move_credentials(cls, data: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
This doesn't feel right to me either, because it's using a deprecated code path to configure credentials for the REST store, but I have to keep this here, at least for a while anyway, for backwards-compatibility purposes.

Ideally, env variables like ZENML_STORE_API_KEY, ZENML_STORE_API_TOKEN and maybe even ZENML_STORE_API_URL should be renamed to ZENML_API_KEY and ZENML_API_TOKEN and processed by the credentials store itself, but that's yet another breaking change that I don't have cycles for right now.

Reply:
Yep makes sense, let's leave it like this for now and hopefully we can refactor/get rid of this path soon!